### PR TITLE
fix(security): #EVAL-480 added resource filter, parameter and unit test DevoirController

### PR DESCRIPTION
**Pull request by simonchaval**
## Describe your changes
Added resource filter on two routes in DevoirController
- @Get("/devoirs/service")
- @Get("/devoir/:idDevoir/moyenne")
- Create unit test case for the new security created.

Added parameter in front end route.

## Issue ticket number and link
[EVAL-480](https://jira.support-ent.fr/browse/EVAL-480)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/ts/models/common/ServiceSnipplet.ts [hunk 1]
```diff
@@ -215,7 +216,7 @@ export class Service {
 
     updateDevoirsService(devoirs, matiere) {
         try {
-            return http.put("/competences/devoirs/service", {
+            return http.put("/competences/devoirs/service?idEtablissement=" + this.id_etablissement, {
```
Comment (lapeyronnie) [inline src/main/resources/public/ts/models/common/ServiceSnipplet.ts:219]: structureId

File: src/main/resources/public/ts/models/teacher/Devoir.ts [hunk 2]
```diff
@@ -364,7 +364,7 @@ export class Devoir extends Model implements IModel {
             try {
                 let allPromise = [];
                 if (moyenne)
-                    allPromise.push(axioshttp.get(this.api.getMoyenneDevoir));
+                    allPromise.push(axioshttp.get(`${this.api.getMoyenneDevoir}?idEtablissement=${this.id_etablissement}`));
```
Comment (lapeyronnie) [inline src/main/resources/public/ts/models/teacher/Devoir.ts:367]: structureId

File: src/main/resources/public/ts/models/teacher/DevoirCollection.ts [hunk 3]
```diff
@@ -104,7 +104,7 @@ export class DevoirsCollection {
                     await devoir.eleves.sync();
                 }
 
-                let url = this.api.done + "?idDevoir=" + devoir.id + "&nbStudents=" + devoir.eleves.length();
+                let url = this.api.done + "&idDevoir=" + devoir.id + "&nbStudents=" + devoir.eleves.length();
```
Comment (lapeyronnie) [inline src/main/resources/public/ts/models/teacher/DevoirCollection.ts:107]: Il ne manque pas l'ajout de l'id de la structure ?

File: src/main/resources/public/ts/models/common/ServiceSnipplet.ts (hunk 1 above)
Comment (Gregprod33) [reply src/main/resources/public/ts/models/common/ServiceSnipplet.ts:219]: Rectifié

File: src/main/resources/public/ts/models/teacher/Devoir.ts (hunk 2 above)
Comment (Gregprod33) [reply src/main/resources/public/ts/models/teacher/Devoir.ts:367]: Rectifié

**Review by lapeyronnie — APPROVED** [review]

**Pull request merged**